### PR TITLE
SWATCH-1014 Configure ActiveMQ

### PR DESCRIPTION
**Pull request by kartikshahc**
https://issues.redhat.com/browse/SWATCH-1014

Test steps:
Set SWATCH_JMS_PRODUCER_ENABLED=true in application.properties and then start the swatch-contract application. You will see something like below in logs indicating that it received the UMB message:
```2023-03-22 13:54:21,427 INFO  [com.red.swa.ContractUMBMessageConsumer] (pool-11-thread-1) {} {
  "action" : "contract-updated",
  "redHatSubscriptionNumber" : "12400374",
  "currentDimensions" : [ {
    "dimensionName" : "test_dim_1",
    "dimensionValue" : "5",
    "expirationDate" : "2023-02-15T00:00:00Z"
  }, {
    "dimensionName" : "test_dim_2",
    "dimensionValue" : "10",
    "expirationDate" : "2023-02-15T00:00:00Z"
  } ],
  "cloudIdentifiers" : {
    "awsCustomerId" : "896801664647",
    "productCode" : "1e1234el8qbwnqimt2wogc5n"
  }
}
```

Note: Since the Partner gateway have changed the structure of response currently, it won't pull any information from API.

### CODE REVIEW

**Review by kahowell — CHANGES_REQUESTED** [review]

File: swatch-contracts/src/main/resources/application.properties [hunk 1]
```diff
@@ -94,3 +94,8 @@ quarkus.rest-client."com.redhat.swatch.clients.rh.partner.gateway.api.resources.
 # configuration properties for subscriptions-sync
 quarkus.rest-client."com.redhat.swatch.clients.swatch.internal.subscription.api.resources.InternalSubscriptionsApi".url=${SWATCH_INTERNAL_SUBSCRIPTION_ENDPOINT}
 quarkus.rest-client."com.redhat.swatch.clients.swatch.internal.subscription.api.resources.InternalSubscriptionsApi".providers=com.redhat.swatch.contract.config.SwatchPskHeaderFilter
+
+quarkus.artemis.url=tcp://localhost:61616
```
[the remark this answers is not in the document]
Comment (kartikshahc) [reply swatch-contracts/src/main/resources/application.properties]: done

File: swatch-contracts/src/main/java/com/redhat/swatch/ContractUMBDummyProducer.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import javax.inject.Inject;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSContext;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/** This class is only to test the ContractUMBMessageConsumer via ActiveMQ JMS */
+@ApplicationScoped
+@Slf4j
+public class ContractUMBDummyProducer {
```
[the remark this answers is not in the document]
Comment (kartikshahc) [reply swatch-contracts/src/main/java/com/redhat/swatch/ContractUMBDummyProducer.java]: done

File: swatch-contracts/src/main/java/com/redhat/swatch/ActiveMQReactiveConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import org.apache.activemq.artemis.jms.client.ActiveMQJMSConnectionFactory;
+import org.apache.activemq.artemis.utils.uri.URISupport;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@ApplicationScoped
+@Slf4j
+public class ActiveMQReactiveConfiguration {
+
+  @ConfigProperty(name = "ACTIVEMQ_BROKER_URL", defaultValue = "vm://0")
+  URI brokerUrl;
```
[the remark this answers is not in the document]
Comment (kartikshahc) [reply swatch-contracts/src/main/java/com/redhat/swatch/ActiveMQReactiveConfiguration.java]: done

File: swatch-contracts/deploy/clowdapp.yaml [hunk 4]
```diff
@@ -47,6 +47,10 @@ parameters:
     value: swatch-tally
   - name: QUARKUS_PROFILE
     value: prod
+  - name: ACTIVEMQ_BROKER_URL
+    value: vm://0
+  - name: ACTIVEMQ_SSL_ENABLED
+    value: 'false'
```
[the remark this answers is not in the document]
Comment (kartikshahc) [reply swatch-contracts/deploy/clowdapp.yaml]: done

File: swatch-contracts/build.gradle [hunk 5]
```diff
@@ -53,6 +53,9 @@ dependencies {
     testAnnotationProcessor libraries["mapstruct-processor"]
     // if you are using mapstruct in test code
     annotationProcessor libraries["lombok-mapstruct-binding"]
+    implementation 'io.quarkus:quarkus-artemis-jms'
+    implementation 'org.apache.activemq:artemis-server:2.28.0'
+    implementation 'org.apache.activemq:artemis-jms-client:2.28.0'
```
[the remark this answers is not in the document]
Comment (kartikshahc) [reply swatch-contracts/build.gradle]: done

**Comment by kahowell** [comment]
Note, test failure is unrelated, so we're going to go ahead and merge :-)

**Pull request merged**